### PR TITLE
JW-521 fixed cloned packets handling

### PR DESCRIPTION
**Pull request by mgreg0r**

### CODE REVIEW

**Comment by mgreg0r** [comment]
Janie, przetestuj proszę

File: dp-core/vr_interface.c [hunk 1]
```diff
@@ -89,19 +89,22 @@ vif_drop_pkt(struct vr_interface *vif, struct vr_packet *pkt, bool input)
...
+    struct vr_packet *pkt = *ppkt;
+
     if (!len)
         return pkt_data(pkt);
 
     if (pkt_head_space(pkt) < len) {
         pkt = vr_pexpand_head(pkt, len - pkt_head_space(pkt));
         if (!pkt)
             return NULL;
+        *ppkt = pkt;
```
Comment (michal-clapinski) [inline dp-core/vr_interface.c]: I don't like this new variable `ppkt`. Couldn't we just go with `*pkt` everywhere?

Comment (sodar) [reply dp-core/vr_interface.c]: I think we could go with using `*pkt`.

File: windows/vr_host.c [hunk 2]
```diff
@@ -549,10 +549,10 @@ win_pexpand_head(struct vr_packet *pkt, unsigned int hspace)
 
     PNET_BUFFER nb = NET_BUFFER_LIST_FIRST_NB(new_nbl);
     if (nb == NULL)
-        return NULL;
+        goto cleanup;
 
     if (NdisRetreatNetBufferDataStart(nb, hspace, 0, NULL) != NDIS_STATUS_SUCCESS)
-        return NULL;
+        goto cleanup;
```
Comment (michal-clapinski) [inline windows/vr_host.c]: Maybe change 539 to `return NULL;` and remove 571.

**Comment by michal-clapinski** [comment]
Very nice :1st_place_medal:

**Comment by mgreg0r** [comment]
Janie, przetestuj proszę

**Pull request merged**